src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+const employeesArray = document.querySelectorAll('li');
+const employeesList = [];
+const ulElement = document.querySelector('ul');
+
+// Перебираем элементы и добавляем объекты в массив
+employeesArray.forEach((employee) => {
+  employeesList.push({
+    name: employee.textContent.trim(),
+    position: employee.dataset.position,
+    salary: Number(employee.dataset.salary.replace('$', '').replace(',', '')),
+    age: employee.dataset.age,
```

Comment:
The age is currently being stored as a string. Consider converting it to a number for consistency and potential future calculations.